test/lib/match.test.js

```diff
@@ -328,3 +328,23 @@ test('file extension mapping', testRule({
     },
   ],
 }));
+
+test('custom message', testRule({
+  valid: [
+    {
+      code,
+      filename: '/foo/bar/test.txt',
+      options: [/^test(?:\..*)?$/],
```

Comment:
Testing both valid and invalid cases with the same option values would be more useful. Can you add the same message string here? The rest LGTM.